packages/contracts/src/contracts/current/forwarder/MixinForwarderCore.sol

```diff
@@ -0,0 +1,87 @@
+pragma solidity ^0.4.21;
+pragma experimental ABIEncoderV2;
+
+import "../protocol/Exchange/Exchange.sol";
+import "../utils/SafeMath/SafeMath.sol";
+import { WETH9 as EtherToken } from "../tokens/WETH9/WETH9.sol";
+import "../tokens/ZRXToken/ZRXToken.sol";
+import "../utils/LibBytes/LibBytes.sol";
+
+contract MixinForwarderCore is
+    LibOrder,
+    LibBytes,
+    SafeMath
+{
+    uint16  constant public PERCENTAGE_DENOMINATOR = 10000; // 9800 == 98%, 10000 == 100%
+    uint16  constant public MAX_FEE = 1000; // 10%
+    uint16  constant public ALLOWABLE_EXCHANGE_PERCENTAGE = 9800; // 98%
+    uint256 constant MAX_UINT = 2 ** 256 - 1;
+
+    // Revert Strings
+    string constant ERROR_FAILED_TO_FILL_ALL_ORDERS = "FAILED_TO_FILL_ALL_ORDERS";
+    string constant ERROR_UNACCEPTABLE_THRESHOLD = "UNACCEPTABLE_THRESHOLD";
+    string constant ERROR_INVALID_INPUT = "INVALID_INPUT";
+
+    Exchange EXCHANGE;
+    EtherToken ETHER_TOKEN;
+    ZRXToken ZRX_TOKEN;
+    mapping (address => uint256) public balanceOf; // Balance of ZRX tokens
```

Comment:
I could go either way on this. It's about 5000-6000 gas to save the remaining ZRX. The acceptance threshold only permits a max 2% fee abstraction. 

2% of the original order fee isn't much at all when the order is low transaction value. If we assume relayers go with a fee percentage of 1% of the order, then its 2% of 1% of the token order.  Around $5000 TTV it could become whole ZRX.

There's also a UX problem of discovering that you have a balance left in this contract. 

Perhaps it makes sense as a future feature once we see how people are using this on mainnet?